docs/portal.md

```diff
@@ -459,6 +459,82 @@ for vocabulary_name in common_vocabularies:
     assert vocabulary_name in vocabulary_names
 ```
 
+(portal-add-catalog-indexes-example)=
+
+## Add catalog indexes
+
+To add indexes to the portal catalog, use {meth}`api.portal.add_catalog_indexes`.
+This function returns a list of the names of the indexes that were added.
+
+The following collection of code snippets demonstrate how to add indexes and either use default logging, to skip reindexing, or to use a customer logger.
+
+```python
+from plone import api
+
+# Add a single field index
+api.portal.add_catalog_indexes([('my_custom_field', 'FieldIndex')])
+
+# Add multiple indexes with different types
+indexes_to_add = [
+    ('text_content', 'ZCTextIndex'),
+    ('tags', 'KeywordIndex')
+]
+api.portal.add_catalog_indexes(indexes_to_add)
+
+# Add indexes without reindexing
+api.portal.add_catalog_indexes([('quick_field', 'FieldIndex')], reindex=False)
+```
```

Comment:
Why did you remove the doctest that you added earlier?

```suggestion

% invisible-code-block: python
%
% # Verify the indexes were added to the catalog
% catalog = api.portal.get_tool('portal_catalog')
% self.assertIn('custom_field', catalog.indexes())
% self.assertIn('review_date', catalog.indexes())
% self.assertIn('modified_by', catalog.indexes())
% self.assertIn('creation_user', catalog.indexes())

```

Reply:
i actually wrote few things from scratch, during adding `ZCTextIndex` handling. Might have been omitted doing it! 😅